### DISCUSSION TITLE
How to create a layout that doesn't take up all vertical space?

### DISCUSSION THREAD

**Discussion by tiktuk** [Textualize/rich#3594] [Q&A]
Hi,

Thanks for the great work on the library!

One question:

I would like to create a simple layout with two panels side by side. They should each fit their content in height, and half of the terminal width.

However, when I them to a layout, their height is expanded to the entire height of the terminal.

I found the `.expand = True` parameter but that is only for width. And if I force them with `.height = 10` they still get placed in the top of the available space, just with empty space below.

So is there a way to prevent the layout from filling the screen?

Test code:

<img width="1530" alt="Screenshot 2024-12-25 at 00 17 38" src="https://github.com/user-attachments/assets/863a98a5-a5f6-4d44-a1b3-786cac0f238f" />

**Comment by tiktuk** [comment]
Oh, I forgot to add my sample code:

```python

from rich import print
from rich.layout import Layout
from rich.panel import Panel

layout = Layout()

p1 = Panel("Hello " * 75)
p2 = Panel("World " * 100)

p1.expand = True  # Does only apply to width
p2.expand = True

p1.height = 10

layout.split_row(
    Layout(p1),
    Layout(p2)
)

layout.size = 15

print(layout)

```

**Comment by willmcgugan** [comment]
I think the only way to do that would be to set the panel height to `console.size.height`.

Consider using Textual if you want more advanced layout than that.
